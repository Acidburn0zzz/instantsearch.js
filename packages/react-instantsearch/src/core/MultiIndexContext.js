import React, {PropTypes, Component, Children} from 'react';

class MultiIndexContext extends Component {
  getChildContext() {
    return {
      multiIndexContext: {
        targettedIndex: this.props.indexName,
      },
    };
  }

  render() {
    const childrenCount = Children.count(this.props.children);
    const {Root, props} = this.props.root;
    if (childrenCount === 0)
      return null;
    else
      return <Root {...props}>{this.props.children}</Root>;
  }
}

MultiIndexContext.propTypes = {
  // @TODO: These props are currently constant.
  indexName: PropTypes.string.isRequired,
  children: PropTypes.node,
  root: PropTypes.shape({
    Root: React.PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.func,
    ]),
    props: PropTypes.object,
  }).isRequired,
};

MultiIndexContext.childContextTypes = {
  // @TODO: more precise widgets manager propType
  multiIndexContext: PropTypes.object.isRequired,
};

export default MultiIndexContext;
