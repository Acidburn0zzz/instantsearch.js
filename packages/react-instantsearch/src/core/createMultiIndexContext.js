import React, {Component, PropTypes} from 'react';
import MultiIndexContext from './MultiIndexContext';

/**
 * Creates a specialized root MultiIndexContext component. It accepts
 * a specification of the root Element.
 * @param {object} root - the defininition of the root of an MultiIndexContext sub tree.
 * @returns {object} a MultiIndexContext root
 */
export default function createInstantSearch(root) {
  return class CreateInstantSearch extends Component {
    static propTypes = {
      children: React.PropTypes.oneOfType([
        React.PropTypes.arrayOf(React.PropTypes.node),
        React.PropTypes.node,
      ]),
      indexName: PropTypes.string.isRequired,
    };

    render() {
      return (
        <MultiIndexContext
          indexName={this.props.indexName}
          root={root}
          children={this.props.children}
        />
      );
    }
  };
}
