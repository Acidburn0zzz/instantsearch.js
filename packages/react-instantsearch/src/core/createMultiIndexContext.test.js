/* eslint-env jest, jasmine */
/* eslint-disable no-console */
import React from 'react';
import {shallow} from 'enzyme';

import createMultiIndexContext from './createMultiIndexContext';
import MultiIndexContext from './MultiIndexContext';

describe('createMultiIndexContext', () => {
  const CustomMultiIndexContext = createMultiIndexContext({Root: 'div'});

  it('wraps MultiIndexContext', () => {
    const wrapper = shallow(<CustomMultiIndexContext indexName="name"/>);
    expect(wrapper.is(MultiIndexContext)).toBe(true);
    expect(wrapper.props()).toMatchSnapshot();
  });
});
