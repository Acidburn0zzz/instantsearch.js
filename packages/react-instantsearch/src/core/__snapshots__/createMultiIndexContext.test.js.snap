// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMultiIndexContext wraps MultiIndexContext 1`] = `
Object {
  "children": undefined,
  "indexName": "name",
  "root": Object {
    "Root": "div",
  },
}
`;
